dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -845,53 +845,85 @@ public void saveStep(final WorkflowStep step, final User user) throws DotDataExc
 	}
 
 	@CloseDBIfOpened
-	public  Future<WorkflowStep>  deleteStep(final WorkflowStep step, final User user) throws DotDataException {
+	public Future<WorkflowStep> deleteStepHardMode(final WorkflowStep step, final User user)
+			throws DotDataException {
+
+		return this.doDeleteStep(step, user, false, true);
+	}
+
+	@CloseDBIfOpened
+	public Future<WorkflowStep> deleteStep(final WorkflowStep step, final User user)
+			throws DotDataException {
+
+		return this.doDeleteStep(step, user, true, false); // runs async
+	}
 
-		return this.doDeleteStep(step, user, true); // runs async
+	private Future<WorkflowStep> doDeleteStep(final WorkflowStep step, final User user,
+			final boolean async) throws DotDataException {
+		return doDeleteStep(step, user, async, false);
 	}
 
-	private Future<WorkflowStep> doDeleteStep(final WorkflowStep step, final User user, final boolean async) throws DotDataException {
+	/**
+	 * Deletes a given step with all dependencies: actions, actionlets and tasks.
+	 *
+	 * @param step Workflow step to delte
+	 * @param user To who perform the delete operation
+	 * @param async If the delete should run in separated thread
+	 * @param hardMode If validations should be skipped. <strong>Note:</strong> Use this parameter
+	 * with caution, was created for hard deletes from Push publish avoiding all validations.
+	 */
+	private Future<WorkflowStep> doDeleteStep(final WorkflowStep step, final User user,
+			final boolean async, final boolean hardMode) throws DotDataException {

```

Comment:
these two would be DeleteOptions

Reply:
I would annotated with WrapInTransaction since if you call async in false, the delete will be part of the callback